Initium-Odp/src/com/universeprojects/miniup/server/controllers/ViewItemController.java

```diff
@@ -316,27 +317,41 @@ else if (iconUrl!=null && iconUrl.startsWith("http")==false)
 			
 			List<EmbeddedEntity> slotItems = (List<EmbeddedEntity>) item.getProperty("Slotted:slotItems");
 	        List<Map<String, Object>> savedSlot = new ArrayList <Map<String, Object>>();
+			ModifierService mService = new ModifierService(db);
 			
+	        
+	        
 			for (int i = 0; i< (long)field; i++) 
 			{
 			    EmbeddedEntity currentSlot = null;
 			    if(slotItems!=null && i < slotItems.size()) currentSlot = slotItems.get(i);
 			    
 			    if(currentSlot != null)
 			    {
-			        /* A caseswitch will be put here to go through
-			         * all the different gem possibilities and assign
-			         *  a value to slot.Name, slot.Image, etc
-			         */
+			    	Map <String, Object> unsavedSlot = new HashMap <String, Object>();
+			    	
+			    	Object slotName = currentSlot.getProperty("name");
+			    	unsavedSlot.put("slotName", slotName);
+			    	
+			    	Object slotIcon = currentSlot.getProperty("icon");
+			    	unsavedSlot.put("slotName", slotIcon);
+			    	
+			    	List<String> rawModifiers = mService.getFullModifierLines(currentSlot);
+			    	List<String> slotModifiers = mService.getFullModifierLines(rawModifiers);
+			    	unsavedSlot.put("slotModifiers", slotModifiers);
+			    	
+			    	Boolean slotIsEmpty = false;

```

Comment:
you don't have to make a slotIsEmpty variable. Just put false directly into the map. This line is superfluous.